### ISSUE TITLE
MatTable with MatSort behaves strangely with Screen Readers

### ISSUE THREAD

**Issue by JeffKochuk** [angular/components#16968]
#### Bug, feature request, or proposal:
When using <table mat-table ... with a MatSort, the resulting table announces the Sort Button's aria-label as part of the table header.

#### What is the expected behavior?
Only the table header's initial text is announced by screenreaders

#### What is the current behavior?
Every 3rd row says "Name Change Sorting for ..."

#### What are the steps to reproduce?
Check this bug report from JAWS where I initially reported the issue. Firefox seems to work as expected, but JAWS on IE does not.

https://github.com/FreedomScientific/VFO-standards-support/issues/207

Navigate through this table with a screenreader. This table is a sanitized version of the output of MatTable
https://codepen.io/anon/pen/VJYaWq

Expected Header announced: "name"
Actual header announced "Name Change Sorting for..." (Not always, but some rows)

#### What is the use-case or motivation for changing an existing behavior?
This will improve accessibility of MatTable

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
IE + JAWS
Chrome

(Not Firefox)

#### Is there anything else we should know?
Someone from FreedomScientific already proposed a solution:
https://github.com/FreedomScientific/VFO-standards-support/issues/207

**Comment by crisbeto** [comment]
Should be addressed by https://github.com/angular/components/pull/13297.

**Comment by jelbourn** [comment]
Duplicate of #13012
